synapse/push/pusherpool.py

```diff
@@ -97,7 +97,6 @@ def start(self) -> None:
     async def add_or_update_pusher(
         self,
         user_id: str,
-        access_token: Optional[int],
```

Comment:
I'm a little confused at the eventual end-goal for `device_id` -- do we think this is now a required property of all pushers? If so, can we update `add_or_update_pusher` to require it (`device_id: str` without a default)?

Reply:
Ideally yes, *but* there is one place where the device_id/access_token_id is not provided: in the admin API, where an admin can register a user without creating a device/access token for the user, that API can create an email pusher on the fly